sdk-extensions/metric-incubator/src/main/java/io/opentelemetry/sdk/viewconfig/ViewConfig.java

```diff
@@ -0,0 +1,220 @@
+import io.opentelemetry.sdk.metrics.view.View;
+import io.opentelemetry.sdk.metrics.view.ViewBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Enables file based YAML configuration of Metric SDK {@link View}s.
+ *
+ * <p>For example, a YAML file with the following content:
+ *
+ * <pre>
+ *   - selector:
+ *       instrument_name: my-instrument
+ *       instrument_type: COUNTER
+ *       meter_name: my-meter
+ *       meter_version: 1.0.0
+ *       meter_schema_url: http://example.com
+ *     view:
+ *       name: new-instrument-name
+ *       description: new-description
+ *       aggregation: histogram
+ *       attribute_keys:
+ *         - foo
+ *         - bar
+ * </pre>
+ *
+ * <p>Is equivalent to the following configuration:
+ *
+ * <pre>{@code
+ * SdkMeterProvider.builder()
+ *     .registerView(
+ *         InstrumentSelector.builder()
+ *             .setInstrumentName("my-instrument")
+ *             .setInstrumentType(InstrumentType.COUNTER)
+ *             .setMeterSelector(
+ *                 MeterSelector.builder()
+ *                     .setName("my-meter")
+ *                     .setVersion("1.0.0")
+ *                     .setSchemaUrl("http://example.com")
+ *                     .build())
+ *             .build(),
+ *         View.builder()
+ *             .setName("new-instrument")
+ *             .setDescription("new-description")
+ *             .setAggregation(Aggregation.histogram())
+ *             .filterAttributes(key -> new HashSet<>(Arrays.asList("foo", "bar")).contains(key))
+ *             .build());
+ * }</pre>
+ */
+public class ViewConfig {
+
+  private ViewConfig() {}
+
+  /**
+   * Load the view configuration YAML from the file and apply it to the {@link
+   * SdkMeterProviderBuilder}.
+   *
+   * @throws ConfigurationException if unable to interpret file contents
+   */
+  @SuppressWarnings("NullAway")
+  public static void registerViews(
+      SdkMeterProviderBuilder meterProviderBuilder, File viewConfigYamlFile) {
+    List<ViewConfigSpecification> viewConfigSpecs = loadViewConfig(viewConfigYamlFile);
+
+    for (ViewConfigSpecification viewConfigSpec : viewConfigSpecs) {
+      assert viewConfigSpec.getSelectorSpecification() != null;
+      assert viewConfigSpec.getViewSpecification() != null;
+      meterProviderBuilder.registerView(
+          toInstrumentSelector(viewConfigSpec.getSelectorSpecification()),
+          toView(viewConfigSpec.getViewSpecification()));
+    }
+  }
+
+  // Visible for testing
+  @SuppressWarnings("unchecked")
+  static List<ViewConfigSpecification> loadViewConfig(File viewConfigYamlFile) {
```

Comment:
Updated from jackson to snakeyaml. Just manually extracting fields from generic list / and map types here. Snakeyaml works with the javabean spec, but not clear if its possible to make it play nice with autovalue builders. Its not pretty, but it works, and is only 17 lines longer than the jackson solution.